DOCS-88: Rewrite TN HA update procedure for 11.3 (#1630)
freenas/freenas-docs: userguide/system.rst

Updating an HA System
~~~~~~~~~~~~~~~~~~~~~

On the
:menuselection:`Dashboard`
of the |ctrlr-term-primary|, click :guilabel:`CHECK FOR UPDATES`. This
button changes to :guilabel:`UPDATES AVAILABLE` when there is an
available update. Clicking the button redirects to
:menuselection:`System --> Update`.
Click :guilabel:`DOWNLOAD UPDATES` to open the
:ref:`Save Configuration <Saving_The_Configuration_File>` dialog.
Another dialog indicates that both |ctrlrs-term| will be updated. Click
:guilabel:`CONTINUE` to start the update.

While the update is in progress, a warning dialog appears when trying to
leave the
:menuselection:`System --> Update`
screen and an animated "System Updating" icon appears in the top section
of the |web-ui|.

> A dialog shows the update progress for both |ctrlrs-term| and when the

Reviewer: This is kind of awkward and not really clear.  The dialog is shown during updates and reboots?  Or after a reboot?

|ctrlr-term-backup| reboots. When the |ctrlr-term-backup| is back online,

> another dialog prompts to :guilabel:`Complete the Upgrade`.

Reviewer: ```suggestion
another dialog asks to :guilabel:`Complete the Upgrade`.
```

.. figure:: images/truenas/system-update-ha-failover.png

> Click :guilabel:`CLOSE` to return to the

Reviewer: This sentence is not really necessary unless it has some qualification ("Click CLOSE if there was some reason to not do a failover...").

:menuselection:`Dashboard`.

> Click :guilabel:`INITIATE FAILOVER`. A dialog warns that this will

Reviewer: This passively tells the user that a dialog warns about services, but it should be more active, and it should warn about that before telling the user to do the failover.
"A failover is needed to continue the update. This will temporarily interrupt network services. If (that is okay), click INITIATE..."

interrupt system services. Confirm the action and click

> :guilabel:`FAILOVER` to continue. This logs out of the |web-ui|, reboots

Reviewer: Non-specific "This". Instead, reword:
"The |ctrl-term-primary| reboots and activates the |ctrlr-term-backup|.

the |ctrlr-term-primary|, and activates the |ctrlr-term-backup|.

The browser shows the |web-ui| login screen when the |ctrlr-term-primary|
is accessible. Log in to the |web-ui| and wait for the
|ctrlr-term-backup| to finish booting. When the |ctrlr-term-backup| is

> online, a dialog prompts to finish the update.

Reviewer: ```suggestion
online, a dialog appears to complete the update.
```

.. figure:: images/truenas/system-update-ha-pending.png

Click :guilabel:`CONTINUE` to complete the update process. To verify

> both |ctrlrs-term| are updated, go to the

Reviewer: ```suggestion
that that the update is complete, go to the
```

:menuselection:`Dashboard`

> and confirm the :guilabel:`Version` is identical for both |ctrlrs-term|.

Reviewer: ```suggestion
and confirm that :guilabel:`Version` is identical on both |ctrlrs-term|.
```

Updating an HA System
~~~~~~~~~~~~~~~~~~~~~

On the
:menuselection:`Dashboard`
of the |ctrlr-term-primary|, click :guilabel:`CHECK FOR UPDATES`. This
button changes to :guilabel:`UPDATES AVAILABLE` when there is an

> available update. Clicking the button redirects to

Reviewer: "redirects" seems less clear than "goes to".

:menuselection:`System --> Update`.

> Click :guilabel:`DOWNLOAD UPDATES` to open the

Reviewer: This is confusing.  The sentence makes it sound like DOWNLOAD UPDATES is only to open the file dialog.  But the dialog is just a side effect, so this should be "After clicking DOWNLOAD UPDATES, a file dialog opens to save the current configuration. Saving the configuration is strongly recommended...

:ref:`Save Configuration <Saving_The_Configuration_File>` dialog.
Another dialog indicates that both |ctrlrs-term| will be updated. Click
:guilabel:`CONTINUE` to start the update.

While the update is in progress, a warning dialog appears when trying to
leave the
:menuselection:`System --> Update`

> screen and an animated "System Updating" icon appears in the top section

Reviewer: "Appears" is a verb. "Is shown" seems more appropriate here:
```suggestion
screen and an animated "System Updating" icon is shown in the top section
```

of the |web-ui|.

A dialog shows the update progress for both |ctrlrs-term|. The
|ctrlr-term-backup| reboots when the update is complete. When the
|ctrlr-term-backup| is back online, a new dialog asks to
:guilabel:`Complete the Upgrade`.

.. figure:: images/truenas/system-update-ha-failover.png

On the
:menuselection:`Dashboard`,

> click :guilabel:`INITIATE FAILOVER` to continue the update. This will

Reviewer: As above, what the sentence says the button does is wildly different from the name of the button. This should be introduced.

temporarily interrupt system services and availability. To start the
failover and interrupt network services, confirm the action and click
:guilabel:`FAILOVER`. The |ctrlr-term-primary| reboots and the
|ctrlr-term-backup| activates.

The browser shows the |web-ui| login screen when the |ctrlr-term-primary|
is accessible. Log in to the |web-ui| and wait for the

> |ctrlr-term-backup| to finish booting. When the |ctrlr-term-backup| is

Reviewer: Is a dialog or status shown when the user is waiting?  How do they know how long it will take?

Author: I'm adding text about the HA status icon, but there really isn't any indication after logging in that the standby controller is rebooting or even that an update is happening.

online, click :guilabel:`CONTINUE` to complete the update process.

.. figure:: images/truenas/system-update-ha-pending.png

> To verify the update is complete,

Reviewer: Missing a verb here...
```suggestion
To verify the update is complete, go to
```

Updating an HA System
~~~~~~~~~~~~~~~~~~~~~

On the
:menuselection:`Dashboard`
of the |ctrlr-term-active|, click :guilabel:`CHECK FOR UPDATES`. This
button changes to :guilabel:`UPDATES AVAILABLE` when there is an
available update. Clicking the button goes to
:menuselection:`System --> Update`.

> Clicking :guilabel:`DOWNLOAD UPDATES` opens a dialog to save the

Reviewer: We are still implying to the user that a button labeled "DOWNLOAD UPDATES" actually does not download updates.
```suggestion
When :guilabel:`DOWNLOAD UPDATES` is clicked, it first gives an
opportunity to save the current system configuration, then available
updates are downloaded.
```

current system configuration. Backing up the system configuration is
strongly recommended before starting the update. Click
:guilabel:`CONTINUE` to start updating both |ctrlrs-term|.

> While the update is in progress, a warning dialog appears when trying to

Reviewer: Is it necessary to tell the user about the dialog if they try to leave that page?  That can just fall under normal error handling which does not need to be described.

leave the
:menuselection:`System --> Update`

> screen and an animated "System Updating" icon is shown in the top

Reviewer: "animated" is not needed.
A "System Updating" icon is shown in the top bar when an update is in progress.

section of the |web-ui|.

> The update progress is shown for both |ctrlrs-term|. The

Reviewer: ```suggestion
Update progress is shown for both |ctrlrs-term|. The
```

|ctrlr-term-standby| reboots when it is finished updating. When the
|ctrlr-term-standby| is back online, the system must

> :ref:`fail over <Failover>` to to finish updating the

Reviewer: Duplicate "to".
```suggestion
:ref:`fail over <Failover>` to finish updating the
```

|ctrlr-term-active|.

.. figure:: images/truenas/system-update-ha-failover.png

To reboot the |ctrlr-term-active| and activate the |ctrlr-term-standby|,
go to the
:menuselection:`Dashboard`
and click :guilabel:`INITIATE FAILOVER` . This will temporarily
interrupt system services and availability. To start the failover,
confirm the action and click :guilabel:`FAILOVER`. The browser logs out
of the |web-ui| and shows the failover status.

The browser shows the |web-ui| login screen when the |ctrlr-term-active|
is accessible. Log in to the |web-ui| and check the
:ref:`HA status icon <HA icon>` in the top toolbar. This icon shows that
HA is unavailable while the |ctrlr-term-standby| reboots. The icon

> updates to show HA is available when the |ctrlr-term-standby| is back

Reviewer: "updates" here is confusing, given the context (system updates).
```suggestion
changes to show HA is available when the |ctrlr-term-standby| is back
```

> online . A dialog appears and asks to finish the update. Click

Reviewer: A bit redundant. Simplified:
```suggestion
online . When the system asks to finish the update, click :guilabel:`CONTINUE`
to complete updating the |ctrlr-term-standby|.
```

:guilabel:`CONTINUE` to finish updating the |ctrlr-term-standby|.

.. figure:: images/truenas/system-update-ha-pending.png

> To verify the update is complete, go to the

Reviewer: Remove a pause by rewording and make this imperative:
```
Verify that the update is complete by going to
:menuselection:`Dashboard`
and confirming that :guilabel:`Version` is the same on
both |ctrlrs-term|.
```
